packages/webpack-cli/lib/utils/cli-flags.js

```diff
@@ -91,8 +91,9 @@ const core = [
         usage: '--config <path to webpack configuration file>',
         alias: 'c',
         type: String,
-        defaultValue: null,
+        defaultValue: [],
```

Comment:
It is still `[]` by default?

Reply:
I think it is because having a `defaultValue` ensures that `ConfigGroup`  will always run even without passing `--config` hence leading to the selection of default configuration file i.e, `webpack.config.js`.
So as of right now if we remove the `defaultValue` option here cli will not load `ConfigGroup` and hence `webpack.config.js` will not be selected by default.

/cc @anshumanv am I right?

Reply:
> It is still [] by default?

Yes we need [] to append all config values because of arg parser https://github.com/webpack/webpack-cli/pull/1765#discussion_r475077654, when we do `webpack -c 1.js -c 2.js` in config group we'll get `{config: ['1.js', '2.js']}`

It is only for arg parsing, not passed to the compiler when no config is supplied

Reply:
> I think it is because having a defaultValue ensures that ConfigGroup will always run even without passing --config hence leading to the selection of default configuration file i.e, webpack.config.js.

Nope @snitin315 all groups are always run 😄

Reply:
>Yes we need [] to append all config values because of arg parser #1765 (comment), when we do webpack -c 1.js -c 2.js in config group we'll get {config: ['1.js', '2.js']}

We don't have `defaultValue: []` for `entry` but it works perfectly as expected. 

>Nope @snitin315 all groups are always run 

interesting I remove the `defaultValue: null`  line and it wasn't selecting `webpack.config.js`

Reply:
I think we need improve arg-parser to avoid this misleading, but we can do it in separate PR

Reply:
> We don't have defaultValue: [] for entry but it works perfectly as expected.

Interesting, will check the exact behaviour during improvement 💯